arangod/Transaction/Methods.cpp

```diff
@@ -409,1386 +409,1607 @@ OperationResult emptyResult(OperationOptions const& options) {
+        _keyBuilder->add(VPackValue(key));
+        value = _keyBuilder->slice();
+      }
+    } else if (value.isObject()) {
+      VPackSlice keySlice = value.get(StaticStrings::KeyString);
+      if (keySlice.isString()) {
+        key = keySlice.stringView();
+      }
+    }
 
-  // append / and key part
-  temp.push_back('/');
-  temp.append(key.data(), key.size());
+    // primary key must not be empty
+    if (key.empty()) {
+      return {TRI_ERROR_ARANGO_DOCUMENT_HANDLE_BAD};
+    }
 
-  builder.add(StaticStrings::IdString, VPackValue(temp));
+    std::pair<LocalDocumentId, RevisionId> lookupResult;
+    Result res = _collection.getPhysical()->lookupKeyForUpdate(&_methods, key,
+                                                               lookupResult);
+    if (res.fail()) {
+      // Error reporting in the babies case is done outside of here.
+      if (res.is(TRI_ERROR_ARANGO_CONFLICT) && !isArray) {
+        TRI_ASSERT(_replicationType != Methods::ReplicationType::FOLLOWER);
+        // if possible we want to provide information about the conflicting
+        // document, so we need another lookup. However, theoretically it is
+        // possible that the document has been deleted by now.
+        auto lookupRes = _collection.getPhysical()->lookupKey(
+            &_methods, key, lookupResult, ReadOwnWrites::yes);
+        TRI_ASSERT(!lookupRes.ok() || lookupResult.second.isSet());
+        _methods.buildDocumentIdentity(_collection, _resultBuilder,
+                                       _trxColl.id(), key, lookupResult.second,
+                                       RevisionId::none(), nullptr, nullptr);
+        auto index = _collection.getPhysical()->primaryIndex();
+        if (index) {
+          return index->addErrorMsg(res, key);
+        }
+      }
+      return res;
+    }
 
-  // _key
-  builder.add(StaticStrings::KeyString, VPackValue(key));
+    std::ignore = _methods.state()->ensureSnapshot();
 
-  // _rev
-  if (rid.isSet()) {
-    char ridBuffer[arangodb::basics::maxUInt64StringSize];
-    builder.add(StaticStrings::RevString, rid.toValuePair(ridBuffer));
-  } else {
-    builder.add(StaticStrings::RevString, std::string_view{});
-  }
+    TRI_ASSERT(lookupResult.first.isSet());
+    TRI_ASSERT(lookupResult.second.isSet());
+    auto [oldDocumentId, oldRevisionId] = lookupResult;
 
-  // _oldRev
-  if (oldRid.isSet()) {
-    builder.add("_oldRev", VPackValue(oldRid.toString()));
-  }
+    _previousDocumentBuilder->clear();
+    res = _collection.getPhysical()->lookupDocument(
+        _methods, oldDocumentId, *_previousDocumentBuilder,
+        /*readCache*/ true, /*fillCache*/ false, ReadOwnWrites::yes);
 
-  // old
-  if (oldDoc != nullptr) {
-    builder.add(StaticStrings::Old, oldDoc->slice());
-  }
+    if (res.fail()) {
+      return res;
+    }
 
-  // new
-  if (newDoc != nullptr) {
-    builder.add(StaticStrings::New, newDoc->slice());
-  }
+    bool excludeFromReplication = _excludeAllFromReplication;
```

Comment:
IMO `excludeFromReplication` will always have the same value as `_excludeAllFromReplication` in a remove operation.
So we can get rid of this local variable here.